_posts/2024-07-10-ansible-yaml-wildfly.adoc

```diff
@@ -0,0 +1,740 @@
+----
+...
+FirstQueue:
+  entries:
+    - 'java:/jms/queue/first'
+  durable: false
+  legacy-entries:
+    - 'java:/jms/legacy/queue/old'
+SecondQueue:
+...
+----
+
+It's already quite nice to be able to express our changes to the subsystem configuration inside a simple text file, but, thanks to Ansible we can go further than that. Currently, the resource settings are somewhat hard-coded in this file, however, we can do better here.
+
+Ansible can easily generate the content of this file using its **templating mechanism**. Which means that we can even abstract part of the configuration and not have all the value hard-coded in the file.
+
+Let's assume, for instance, that FirstQueue is not durable when deployed on _staging_ systems. We can employ a template so that Ansible can create the appropriate configuration depending on the target system. Relying on the internal convention that any staging system as the suffix `'. stating'` in the machine hostname, Ansible be able to change the default value of durable from true to false:
+
+[source, yml]
+----
+wildfly-configuration:
+  subsystem:
+    messaging-activemq:
+      server:
+        default:
+          jms-queue:
+            FirstQueue:
+              entries:
+                - 'java:/jms/queue/first'
+{% if '.staging' in ansible_nodename %}
+              durable: false
+{% endif %}
+              legacy-entries:
+                - 'java:/jms/legacy/queue/old'
+            SecondQueue:
+              entries:
+                - 'java:/jms/queue/second'
+          jms-topic:
+            TheTopic:
+              entries:
+                - topic/TheTopic
+                - java:jboss/exported/topic/TheTopic
+----
+
+While this templating feature is quite powerful, a balance needs to be found when it is leveraged. Generating the entire template, based on rather complex data structure is not advisable, for instance. The Yaml Config file is already a configuration artifact, that can be used as a source of truth.
+
+In short, when designing the way WildFly's setup will be provisioned, it's important to determine what needs to be added directly to the default configuration (`standalone.xml` or `standalone-full.xml`) utilized as a base and what can be parameterized using the Yaml configuration feature, employing or not, the templating functionality of Ansible.
+
+To help make these decisions, here are a few rules of thumb to keep in mind:
+
+* Large alteration of the subsystems (adding one or several, or simply rewriting entirely the default configuration) are most likely easier to achieve by providing a modified base configuration.
+* Small changes to the subsystem configuration, adding a few, straightforward resources are most likely easy enough to implement.
+* Changes in the configuration linked to the target environments can be achieved using the templating feature of Ansible.
+* No matter what, remember the https://en.wikipedia.org/wiki/KISS_principle[KISS principe (Keep It Stupid Simple)].
```

Comment:
s/principe/principle

You're nice; for us mean unprofessional people it's "Keep It Simple, Stupid!"  Your way is better for this kind of article. :)